tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/MockTest.java

```diff
@@ -144,6 +144,19 @@ public void testCoverageWithMocking() throws BallerinaTestException {
         Assert.assertEquals(moduleObj.get("coveragePercentage").toString(), "57.14");
     }
 
+    @Test
+    public void testFunctionMockingInMultipleModulesWithDependencies() throws BallerinaTestException {
+        String msg1 = "function_mocking_with_dependencies.a_dependency\n\t\t[pass] test1\n\n\n\t\t" +
+                "1 passing\n\t\t0 failing\n\t\t0 skipped";
+        String msg2 = "function_mocking_with_dependencies.b_dependent\n\t\t[pass] test1\n\n\n\t\t" +
+                "1 passing\n\t\t0 failing\n\t\t0 skipped";
+        String[] args = mergeCoverageArgs(new String[]{"function-mocking-tests-with-dependencies"});
+        String output = balClient.runMainAndReadStdOut("test", args, new HashMap<>(), projectPath, false);
+        if (!output.contains(msg1) || !output.contains(msg2)) {
+            Assert.fail("Test failed due to function mocking failure in test framework.\nOutput:\n" + output);
```

Comment:
Shall we update the test case to have the expected output in a file and compare in order to be consistent with the new integration test improvements?

Reply:
Added the output files for the test. As per an offline discussion with @Thevakumar-Luheerathan we can change the test case to read from the output file instead of the string in https://github.com/ballerina-platform/ballerina-lang/pull/39562